materialize-mysql/infile.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	stdsql "database/sql"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+const (
+	// Limit the maximum size of buffered data before flushing a batch.
+	batchSizeThreshold = 10 * 1024 * 1024
```

Comment:
This was previously `33554432`, but I am lowering it here. This should still work for large documents, as we don't limit the size of the buffer and just use this limit to account for when the buffer should be drained. As before, a really large document will be serialized into the buffer and then the buffer will immediately be drained. But the lower limit will work better in the common case of small documents to keep the connector memory usage low.